src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      basePath: BASE_PATH + "v1",
+      apiKey: apiKey,
+    });
+    const v2Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v2",
+      apiKey: apiKey,
+    });
+    this.apis = {
+      v1Cast: new CastApiV1(v1Config, undefined, axiosInstance),
+      user: new UserApi(v1Config, undefined, axiosInstance),
+      verification: new VerificationApi(v1Config, undefined, axiosInstance),
+      notifications: new NotificationsApi(v1Config, undefined, axiosInstance),
+      reactions: new ReactionsApi(v1Config, undefined, axiosInstance),
+      v1Follows: new FollowsApiV1(v1Config, undefined, axiosInstance),
+      v2Cast: new CastApiV2(v2Config, undefined, axiosInstance),
+      signer: new SignerApi(v2Config, undefined, axiosInstance),
+      reaction: new ReactionApi(v2Config, undefined, axiosInstance),
+      v2Follow: new FollowApiV2(v2Config, undefined, axiosInstance),
+    };
+  }
+
+  /**
+   * Utility for parsing errors returned by the Merkle API server. Returns true
+   * if the given error is caused by an error response from the server, and
+   * narrows the type of `error` accordingly.
+   */
+  public static isApiErrorResponse(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    error: any
+  ): error is WithRequired<AxiosError<ErrorRes>, "response"> {
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
+  }
+
+  /**
+   * Creates a Signer. See [Neynar documentation](https://docs.neynar.com/reference/create-signer)
+   * for more details.
+   */
+  public async createSigner(): Promise<Signer> {
+    const response = await this.apis.signer.createSigner();
+    return response.data;
+  }
+
+  /**
+   * Fetches an existing Signer. See [Neynar documentation](https://docs.neynar.com/reference/get-signer)
+   * for more details.
+   *
+   */
+  public async fetchSigner(signerUuid: string): Promise<Signer | undefined> {
+    try {
+      const response = await this.apis.signer.signer(signerUuid);
+      return response.data;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Registers a Signer with an fid. See [Neynar documentation](https://docs.neynar.com/reference/register-app-fid)
+   * for more details.
+   */
+  public async registerSigner(
+    signerUuid: string,
+    fid: number,
+    deadline: number,
+    signature: string
+  ): Promise<Signer> {
+    const body: RegisterSignerKeyReqBody = {
+      signer_uuid: signerUuid,
+      app_fid: fid,
+      deadline,
+      signature,
+    };
+    const response = await this.apis.signer.registerSignedKey(body);
+    return response.data;
+  }
+
+  /**
+   * Gets information about an individual cast using neynar v1 API
+   */
+  public async v1FetchCast(
```

Comment:
For methods like these that have both `v1` and `v2` implementations we should document what the difference is and why a user would use one over the other.

(The Neynar team should probably help us in producing this documentation)

I imagine the neynar-specific README should also address this point as well, at the general level

Reply:
This is the only method that's implemented with both and I lean towards just removing it because there's nothing the v1 offers that v2 doesn't have. The only difference is in the response of type Cast, which is slightly different between the v1 and v2 specs. Again, the v2 spec includes all of what is in v1, so there's no reason to use the specific v1 fetch.